projects/ui-components/lf-selection-list/lf-selection-list.component.ts

```diff
@@ -227,12 +227,13 @@ export class LfSelectionListComponent implements AfterViewInit, OnDestroy {
     }
   }
 
-  setSelectedValuesAsync(
+  async setSelectedNodesAsync(
     values: ILfSelectable[],
-    checkForMoreDataCallback: () => Promise<ILfSelectable[] | undefined>
+    checkForMoreDataCallback: () => Promise<ILfSelectable[] | undefined>,
+    maxFetchIterations: number
   ): Promise<ILfSelectable[]> {
     this.selectable.callback = checkForMoreDataCallback;
-    return this.selectable.setSelectedValuesAsync(values, this.listItems).then(() => {
+    return await this.selectable.setSelectedNodesAsync(values, this.listItems, maxFetchIterations).then(() => {
```

Comment:
you can just do 
```ts
await this.selectable.setSelectedNodesAsync(...)
return this.selectable.selectedItems
```
rather than mixing the await and .then syntax